rasa/core/evaluation/marker.py

```diff
@@ -182,12 +189,12 @@ class IntentDetectedMarker(ConditionMarker):
     @staticmethod
     def positive_tag() -> Text:
         """Returns the tag to be used in a config file."""
-        return "intent_detected"
+        return "intent"
 
     @staticmethod
     def negated_tag() -> Optional[Text]:
         """Returns the tag to be used in a config file for the negated version."""
-        return "intent_not_detected"
+        return "not_intent"
```

Comment:
"not_intent" doesn't give the same meaning as "intent_not_detected".  It sounds like "tag all intents that are not this intent" rather than "tag dialogues when the intent never occurs".

Reply:
"not_intent: bla" is true whenever "intent: bla" is not true - it is exactly what it sounds like

Reply:
for "never" and "occurs" we have the operators

Reply:
.... ok, then we'll switch back to the old ones

Reply:
but to avoid confusion: "intent_not_detected" does not mean "tag dialogues when the intent never occurs" ... the marker "intent_not_detected : bla " applies to every event in a conversation where the current action is *not* a user uttered event with "bla" as intent. We're not tagging dialogues per se, but we output statistics on per-session level and the overall statistics over all sessions. So if we want to "tag dialogues when..." , we could load the per-session results and look up for which sessions there is at least one relevant event logged for this marker. (Or, if we want to make sure we log this only once, we add an "at_least_once" marker, for which only one relevant event will be logged, namely the first event).

Reply:
Okay I see. Then the meaning of the original label changed. It's fine as long as we're clear about what it means. As you say, people can use "NEVER" for the other meaning.

Reply:
By original you mean the previous implementation, right? Yes, the "intent_detected" is now defined exactly as in weizenbot (https://www.notion.so/rasa/Weizenbot-Conditions-Docs-c268cdaa34cd4ec9a68d067620c4f4d7#8739eae05a684e58a73e7d5f3d5f856c) and afaik "intent_not_detected"  does not exist there (We added those "not" versions only because they were in the original proposal). In the previous implementation, there was an implicit "at_least_once" in front of every atomic condition (i.e. a negated "never" in weizenbot).